### DISCUSSION TITLE
Testing an OPA policy which has built-in function http.send

### DISCUSSION THREAD

**Discussion by Ronnie-personal** [open-policy-agent/community#421] [OPA and Rego] [answered]
Hi There,

What would be the mock up options for testing a policy which utilizes OPA built-in function http.send?
Does OPA offer out of box mock up for its built-in http.send function?

Thank you so much!

**Comment by anderseknert** [comment, marked as answer]
Hi Ronnie!

Yep, see https://www.openpolicyagent.org/docs/latest/policy-testing/#data-and-function-mocking

**Reply by Ronnie-personal** [reply]
Excellent! Thanks for sharing the document link!
